src/instructlab/configuration/init.py

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from os import listdir
+from os.path import basename, dirname, exists, splitext
+
+# Third Party
+from git import GitError, Repo
+import click
+
+# First Party
+from instructlab import config, utils
+
+
+@click.command()
+@click.option(
+    "--interactive/--non-interactive",
+    default=True,
+    show_default=True,
+    help="Initialize the environment assuming defaults.",
+)
+@click.option(
+    "--model-path",
+    type=click.Path(),
+    default=config.DEFAULT_MODEL_PATH,
+    show_default=True,
+    help="Path to the model used during generation.",
+)
+@click.option(
+    "--taxonomy-base",
+    default=config.DEFAULT_TAXONOMY_BASE,
+    show_default=True,
+    help="Base git-ref to use when listing/generating new taxonomy.",
+)
+@click.option(
+    "--taxonomy-path",
+    type=click.Path(),
+    default=config.DEFAULT_TAXONOMY_PATH,
+    show_default=True,
+    help=f"Path to {config.DEFAULT_TAXONOMY_REPO} clone.",
+)
+@click.option(
+    "--repository",
+    default=config.DEFAULT_TAXONOMY_REPO,
+    show_default=True,
+    help="Taxonomy repository location.",
+)
+@click.option(
+    "--min-taxonomy",
+    is_flag=True,
+    help="Shallow clone the taxonomy repository with minimum size. "
+    "Please do not use this option if you are planning to contribute back "
+    "using the same taxonomy repository. ",
+)
+def init(
+    interactive,
+    model_path,
+    taxonomy_path,
+    taxonomy_base,
+    repository,
+    min_taxonomy,
+):
+    """Initializes environment for InstructLab"""
+    clone_taxonomy_repo = True
+    if interactive:
+        if exists(config.DEFAULT_CONFIG):
+            overwrite = click.confirm(
+                f"Found {config.DEFAULT_CONFIG} in the current directory, do you still want to continue?"
+            )
+            if not overwrite:
+                return
+        click.echo(
+            "Welcome to InstructLab CLI. This guide will help you to setup your environment."
+        )
+        click.echo(
+            "Please provide the following values to initiate the "
+            "environment [press Enter for defaults]:"
+        )
+        taxonomy_path = utils.expand_path(
+            click.prompt("Path to taxonomy repo", default=taxonomy_path)
+        )
+
+    try:
+        taxonomy_contents = listdir(taxonomy_path)
```

Comment:
I would prefer this remained `os.listdir()` to reduce the diff when making sure accidental changes weren't included or newer changes lost by accident

(I know this was in an older comment, so apologies if this is a duplicate. I don't know where the old comment went)